### ISSUE TITLE
Problems with the Deca training dataset

### ISSUE THREAD

**Issue by nainsheng** [YadiraF/DECA#195]
Hello author, I am currently studying this paper and debugging the code. The demo has been successfully debugged, but the details of the training code are not clear, as shown in the following part:
b. Prepare label
FAN to predict 68 2D landmarks
face_segmentation to get skin mask

I would like to ask how this part should be operated, can you give more details?

**Comment by xuduo18311199384** [comment]
Train Datas include "image.png + landmarks.npy(FAN detect) + segmentation.npy(face_segmentation detect)"

**Comment by nainsheng** [comment]
@xuduo18311199384 
Thank you for your reply. The data related to FAN in the first part has been processed, but we encountered a problem in the facial segmentation step. Firstly, this link needs to install the caffe framework, but many methods have been used but cannot be successfully installed. Secondly, I noticed that I still need Vggface2_train_list_max_normal_100_ring_5_1serial.npy files. Can you give me some prompts? Thank you very much for your help.

**Comment by xuduo18311199384** [comment]
1. face_segmentation part. You must install caffe(with gpu).

**Comment by nainsheng** [comment]
Thank you for your reply. I have completed facial segmentation. What should I do about Vggface2_train_list_max_normal_100_ring_5_1serial.npy? Is it the best 5 facial images for each candidate in the dataset?

**Comment by xuduo18311199384** [comment]
The follow picture should be made by Vggface2_train_list_max_normal_100_ring_5_1serial.npy
each candidate : image/225/0000383.png, 68kpts/225/0000383.npy, face_segmental/225/0000383.npy, 
![2023-08-03 16-10-51 的屏幕截图](https://github.com/yfeng95/DECA/assets/30895764/e37a0696-92d2-4bae-82ce-6a4f861d50e7)
[shape_consistent_loss & detail_loss] require images >= 3 for each candidate

**Comment by anushka17agarwal** [comment]
Hey, I had a doubt. In the vox.py file (Line 126- 128)
`        images_array = torch.from_numpy(np.array(images_list)).type(dtype = torch.float32) #K,224,224,3
        kpt_array = torch.from_numpy(np.array(kpt_list)).type(dtype = torch.float32) #K,224,224,3
        mask_array = torch.from_numpy(np.array(mask_list)).type(dtype = torch.float32) #K,224,224,3`
The shape of kpt_array is 224, 224, 3 , but the face alignment library returns a shape of (62, 3). So, how exactly were these keypoints transformerd

**Comment by zhumeichoubao** [comment]
> Hey, I had a doubt. In the vox.py file (Line 126- 128) ` images_array = torch.from_numpy(np.array(images_list)).type(dtype = torch.float32) #K,224,224,3 kpt_array = torch.from_numpy(np.array(kpt_list)).type(dtype = torch.float32) #K,224,224,3 mask_array = torch.from_numpy(np.array(mask_list)).type(dtype = torch.float32) #K,224,224,3` The shape of kpt_array is 224, 224, 3 , but the face alignment library returns a shape of (62, 3). So, how exactly were these keypoints transformerd

I have the same question, how do you solve it. Can you give me some prompts? Thank you very much for your help.

**Comment by mattisdietrich** [comment]
I think it is done in (Line 102/105 if not changed) `cropped_kpt= np.dot(tform.params, np.hstack([kpt, np.ones([kpt.shape[0],1])]).T).T and then normalized to 224,224,3.`  The tform.params are defined in `self.crop(image,kpt)`

**Comment by mattisdietrich** [comment]
@zhumeichoubao @anushka17agarwal I think it is done in (Line 102/105 if not changed) `cropped_kpt=np.dot(tform.params, np.hstack([kpt, np.ones([kpt.shape[0],1])]).T).T `  and then normalized to 224,224,3. The tform.params are defined in `self.crop(image,kpt)` before, where you can look for detail

**Comment by jylovec** [comment]
> The follow picture should be made by Vggface2_train_list_max_normal_100_ring_5_1serial.npy each candidate : image/225/0000383.png, 68kpts/225/0000383.npy, face_segmental/225/0000383.npy, ![2023-08-03 16-10-51 的屏幕截图](https://private-user-images.githubusercontent.com/30895764/258099212-e37a0696-92d2-4bae-82ce-6a4f861d50e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NzQ0NDIsIm5iZiI6MTcwNTY3NDE0MiwicGF0aCI6Ii8zMDg5NTc2NC8yNTgwOTkyMTItZTM3YTA2OTYtOTJkMi00YmFlLTgyY2UtNmE0Zjg2MWQ1MGU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDE0MjIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2MmI5Zjk4OWE1YmNkMzkxMGVlMzNmMDQ4ZjI2MjE1NDk0MmFiNzM3M2IzY2RmMjkyMjU0ZWRhMWM1ZTMxNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gusNtYJabJezXOAs_hoLDJW89kawjJltcNHxkwhZMCI) [shape_consistent_loss & detail_loss] require images >= 3 for each candidate

why [shape_consistent_loss & detail_loss] require images >= 3 for each candidate？I would appreciate it if you reply me。

**Comment by emlcpfx** [comment]
Were you able to train the model?
